### PR TITLE
80 erstes deep learning modell

### LINKED ISSUE #80: Erstes Deep Learning Modell
- [x] Nutzen von segmentierten Daten

Wenn möglich: 2D-CNN auf Spectral Map zu Daten trainiert, siehe Research Notizen [Spectrogram Maps](https://github.com/Sensor-Based-Activity-Recognition/docs/blob/main/recherche/Preprocessing%20%26%20Features.md#deep-learning)

- [x] Spektrogramm Stage hinzufügen (Output: 1 Image oder Matrix pro Segment, Input: Segments)
- [x] Pytorch Dataset Class
- [x] Abstract new dataset class
- [x] run on GPU
- [x] optimize params
- [x] delete ipynb
- [x] CPU auslastung bei 100%, DataLoader?
- [ ] Merge Branch into Main

### PR DESCRIPTION
**Pull request by florinbarbisch**

### CODE REVIEW

**Comment by ghost** [comment]
now we just need metrics

**Comment by gabrieltorresgamez** [comment]
> now we just need metrics

who tf is u

**Comment by ghost** [comment]
# RUN: CNN, 2023-05-20 15-58-41

## Hyperparameters

| Path        | Param                               | main                    | workspace   |
|-------------|-------------------------------------|-------------------------|-------------|
| params.yaml | dvclive.data.batch_size             | -                       | 64          |
| params.yaml | dvclive.data.num_workers            | -                       | 0           |
| params.yaml | dvclive.data.output_size            | -                       | 6           |
| params.yaml | dvclive.data.params.columns         | numeric                 | -           |
| params.yaml | dvclive.data.train_val_split        | -                       | 0.8         |
| params.yaml | dvclive.data.type                   | Sklearn_Segments        | NDArray     |
| params.yaml | dvclive.model                       | AdaBoostStumpClassifier | CNN         |
| params.yaml | dvclive.model_hparams.learning_rate | 0.5                     | 0.05        |
| params.yaml | dvclive.model_hparams.momentum      | -                       | 0.9         |
| params.yaml | dvclive.model_hparams.n_estimators  | 200                     | -           |
| params.yaml | dvclive.model_hparams.num_epochs    | -                       | 15          |
| params.yaml | dvclive.model_hparams.random_state  | 42                      | -           |
| params.yaml | dvclive.model_hparams.weight_decay  | -                       | 0.0005      |
| params.yaml | dvclive.seed                        | 42                      | 40          |
| params.yaml | train_test_split_ratio02.type       | measurement             | segment     |

## Metrics

| Path                 | Metric           | main    | workspace   | Change   |
|----------------------|------------------|---------|-------------|----------|
| dvclive/metrics.json | epoch            | -       | 15          | -        |
| dvclive/metrics.json | step             | -       | 1275        | -        |
| dvclive/metrics.json | test.epoch.acc   | 0.70765 | 0.89912     | 0.19146  |
| dvclive/metrics.json | test.epoch.f1    | 0.66619 | 0.88909     | 0.2229   |
| dvclive/metrics.json | test.step.acc    | -       | 0.89912     | -        |
| dvclive/metrics.json | test.step.f1     | -       | 0.88909     | -        |
| dvclive/metrics.json | train.epoch.acc  | 0.72471 | 0.90192     | 0.1772   |
| dvclive/metrics.json | train.epoch.f1   | 0.68623 | 0.88443     | 0.1982   |
| dvclive/metrics.json | train.epoch.loss | -       | 0.25035     | -        |
| dvclive/metrics.json | train.step.acc   | -       | 0.9375      | -        |
| dvclive/metrics.json | train.step.f1    | -       | 0.92125     | -        |
| dvclive/metrics.json | train.step.loss  | -       | 0.17694     | -        |
| dvclive/metrics.json | val.epoch.acc    | -       | 0.88708     | -        |
| dvclive/metrics.json | val.epoch.f1     | -       | 0.8804      | -        |
| dvclive/metrics.json | val.epoch.loss   | -       | 0.28894     | -        |
| dvclive/metrics.json | val.step.acc     | -       | 0.88708     | -        |
| dvclive/metrics.json | val.step.f1      | -       | 0.8804      | -        |
| dvclive/metrics.json | val.step.loss    | -       | 0.28894     | -        |

## Plots

### Confusion Matrix

Confusion Matrix

File: src/model_runner.py [hunk 1]
```diff
@@ -66,26 +69,22 @@ def get_model(model_name, config):
...
-    datamodule = DataLoaderSklearn_Tabular(
-        config.data.get("params"), input_filename_data, input_filename_train_test_split
-    )
-elif config.data["type"] == "Sklearn_Segments":
-    datamodule = DataLoaderSklearn_Segments(
-        config.data.get("params"), input_filename_data, input_filename_train_test_split
-    )
-else:
-    raise NotImplementedError(f"Datamodule {config.type} not implemented")
+datamodule_class = {
```
Comment (yvokeller) [inline src/model_runner.py:72]: clean 👍🏻

**Review by yvokeller — APPROVED** [review]
nicely done

**Comment by ghost** [comment]
# RUN: CNN, 2023-05-21 17-49-40

## Hyperparameters

| Path   | Param   | main   | workspace   |
|--------|---------|--------|-------------|

## Metrics

| Path   | Metric   | main   | workspace   | Change   |
|--------|----------|--------|-------------|----------|

## Plots

### Confusion Matrix

Confusion Matrix
